fiat/prices.go

```diff
@@ -37,6 +37,22 @@ type PriceSource struct {
 	impl fiatBackend
 }
 
+// PriceSourceConfig is a struct holding various config options used for
+// initialising a new PriceSource.
+type PriceSourceConfig struct {
```

Comment:
Add a validation method now that we're pulling this out into a separate function? 

eg, coincap has to have granularity, custom backend must have some price points etc